app/renderer/components/Inspector/HighlighterRects.js

```diff
@@ -1,20 +1,152 @@
+          }
+        };
+        elements = elements.concat(this.newCoordinates(elementsByOverlap[key], key));
+      }
+      elements.push(element);
+    }
+
+    // Remove first element as it is an XML document node
+    elements.shift();
+
+    return elements;
+  }
+
+  // Create a new object for each element and determine its properties
+  findElementProps (element, prevElement, elements, overlaps) {
+    if (!element) {
+      return {};
+    }
+    const { scaleRatio, containerEl } = this.props;
+    const {x1, y1, x2, y2} = parseCoordinates(element);
+    const xOffset = this.highlighterXOffset || 0;
+    const posOffset = 0.03 * containerEl.offsetWidth;
+    const obj = {
+      type: RENDER.CENTROID,
+      element,
+      parent: prevElement,
+      properties: {
+        left: x1 / scaleRatio + xOffset || 0,
+        top: y1 / scaleRatio || 0,
+        width: (x2 - x1) / scaleRatio || 0,
+        height: (y2 - y1) / scaleRatio || 0,
+        centerX: ((Math.round(x1 + ((x2 - x1) / 2)) / scaleRatio) + xOffset - posOffset) || 0,
+        centerY: ((Math.round(y1 + ((y2 - y1) / 2)) / scaleRatio) - posOffset) || 0,
+        path: element.path,
+        keyCode: null,
+        container: false,
+        accessible: element.attributes.accessible
+      }
+    };
+    const coordinateName = `${obj.properties.centerX}.${obj.properties.centerY}`;
+    obj.properties.container = this.isElementContainer(obj, elements);
+    elements.push(obj);
+
+    overlaps[coordinateName] ?
+      overlaps[coordinateName].push(obj)
+      :
+      overlaps[coordinateName] = [obj];
+
+    if (element.children) {
+      for (let childEl of element.children) {
+        this.findElementProps(childEl, element, elements, overlaps);
+      }
+    }
+
+    return overlaps;
+  }
+
+  isElementContainer (element1, elements) {
+    for (let element2 of elements) {
+      if (element2.element !== element1.element
+        && this.isElementOverElement(element1.properties, element2.properties)
+        && !this.isAncestor(element1.parent, element2.element, elements)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  isElementOverElement (element1, element2) {
+    return element1.left <= element2.left
+        && element2.width <= element1.width
+        && element1.top >= element2.top
+        && element2.height <= element1.height;
+  }
+
+  isAncestor (parent, checkElement, elements) {
+    if (elements.length > 0) {
+      while (parent !== null) {
+        if (parent === checkElement) { return true; }
+
+        for (let elem of elements) {
+          if (elem.element === parent) { parent = elem.parent; }
+        }
+      }
+    }
+    return false;
+  }
+
+  // Generate new coordinates for overlapping elements
+  newCoordinates (elements, key) {
+    const { containerEl } = this.props;
+    const radius = 0.06 * containerEl.offsetWidth;
+    const steps = elements.length;
+    for (let step = 0; step < steps; step++) {
```

Comment:
const; also if we are just doing a simple loop like this it's ok to use a counter variable like `i`

but more broadly, why not just do `for (const el of elements)`? typically we prefer that kind of iteration over for loops with counters.

Reply:
The reason why I used a loop with counters is just because I needed to use those counters when determining the position of each element overlap. I couldn't find any other way to accomplish that without using for loop counters